src/test/ReactComponentTreeTestUtils.js

```diff
@@ -62,9 +63,21 @@ function expectTree(rootID, expectedTree, parentPath = '') {
   }
   if (expectedTree.text !== undefined) {
     expectEqual(text, expectedTree.text, 'text');
+    expectEqual('' + element, expectedTree.text, 'element.toString()');
   } else {
     expectEqual(text, null, 'text');
   }
+  if (expectedTree.element !== undefined) {
+    // TODO: Comparing elements makes tests run out of memory on errors.
```

Comment:
:( yeah I've gotten this sort of thing a few times. Maybe something about circular references? If it's easy to repro we should file a jest bug.